### PR TITLE
feat(dividends): implement dividend creation

### PR DESCRIPTION
**Pull request by monitz87**

### CODE REVIEW

File: src/api/procedures/configureDividendDistribution.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  } = this;
+
+  return {
+    identityRoles: [
+      { type: RoleType.TokenCaa, ticker },
+      { type: RoleType.PortfolioCustodian, portfolioId: portfolioToPortfolioId(portfolio) },
+    ],
+    signerPermissions: {
+      transactions: [TxTags.capitalDistribution.Distribute],
+      tokens: [],
```
Comment (VictorVicente) [inline src/api/procedures/configureDividendDistribution.ts:210]: I think you need Token permission. Please check.

File: src/api/procedures/__tests__/initiateCorporateAction.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+        defaultTaxWithholding,
+        taxWithholdings,
+      });
+    } catch (error) {
+      err = error;
+    }
+
+    expect(err.message).toBe('Too many target Identities');
+    expect(err.data).toEqual({
+      maxTargets: 1000,
```
Comment (VictorVicente) [inline src/api/procedures/__tests__/initiateCorporateAction.ts:187]: This will fail when you start fetching this value from the chain

**Review by VictorVicente — COMMENTED** [review]
Excellent job!
I really like the `isCodec` typeguard.

**Review by VictorVicente — CHANGES_REQUESTED** [review]
There are some code smells to fix too.

File: src/api/procedures/__tests__/initiateCorporateAction.ts (hunk 2 above)
Comment (monitz87) [reply src/api/procedures/__tests__/initiateCorporateAction.ts:187]: We'll have to change the test anyway to mock the value being returned

File: src/api/procedures/configureDividendDistribution.ts (hunk 1 above)
Comment (monitz87) [reply src/api/procedures/configureDividendDistribution.ts]: For this you do, but not for creating the CA

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**